OmsAgent/omsagent.py

```diff
@@ -46,100 +49,279 @@
+    try:
+        hutil = parse_context(operation)
+        if is_vm_supported_for_extension():
+            exit_code = operations[operation](hutil)
+        else:
+            log_and_exit(hutil, operation, 51, 'Unsupported operation system')
+
+        if exit_code is not 0:
+            message = (operation + ' failed with exit code {0}').format(exit_code)
+
+    except ValueError as e:
+        exit_code = 11
+        message = (operation + ' failed with error: {0}').format(e.message)
+
+    except Exception as e:
+        exit_code = 1
+        message = (operation + ' failed with error: {0}, {1}').format(e, traceback.format_exc())
+
+        if "LINUXOMSAGENTEXTENSION_ERROR_MULTIPLECONNECTIONS" in str(e):
+            exit_code = 10
+            message = (operation + ' failed with error: {0}').format(e.message)
+
+    # Finish up and log messages
+    log_and_exit(hutil, operation, exit_code, message)
 
 
 def parse_context(operation):
     hutil = Util.HandlerUtility(waagent.Log, waagent.Error)
     hutil.do_parse_context(operation)
     return hutil
-    
-    
-def dummy_command(operation, status, msg):
-    hutil = parse_context(operation)
-    hutil.do_exit(0, operation, status, '0', msg)
 
 
-def getbundlefilename():
-    file_name = BundleFileNameTemplate.format('x86')
-    is_64bits = sys.maxsize > 2**32
-    if is_64bits:
-        file_name = BundleFileNameTemplate.format('x64')
-    return file_name
+def is_vm_supported_for_extension():
+    '''
+    Returns for platform.linux_distribution() vary widely in format, such as '7.3.1611' returned
+    for a machine with CentOS 7, so the first provided digits must match
+    '''
+    supported_dists = {'redhat' : ('5', '6', '7'), # CentOS
+                       'centos' : ('5', '6', '7'), # CentOS
+                       'red hat' : ('5', '6', '7'), # Oracle, RHEL
+                       'oracle' : ('5', '6', '7'), # Oracle
+                       'debian' : ('6', '7', '8'), # Debian
+                       'ubuntu' : ('12.04', '14.04', '15.04', '15.10', '16.04'), # Ubuntu
+                       'suse' : ('11', '12') #SLES
+    }
+
+    try:
+        vm_dist, vm_ver, vm_id = platform.linux_distribution()
+    except AttributeError:
+        vm_dist, vm_ver, vm_id = platform.dist()
+
+    vm_supported = False
+
+    # Find this VM distribution in the supported list
+    for supported_dist in supported_dists.keys():
+        if not vm_dist.lower().startswith(supported_dist):
+            continue
+
+        # Check if this VM distribution version is supported
+        vm_ver_split = vm_ver.split('.')
+        for supported_ver in supported_dists[supported_dist]:
+            supported_ver_split = supported_ver.split('.')
+
+            vm_ver_match = True
+            for idx, supported_ver_num in enumerate(supported_ver_split):
+                try:
+                    vm_ver_num = vm_ver_split[idx]
+                except IndexError:
+                    vm_ver_match = False
```

Comment:
If an IndexError is thrown, that means `supported_ver` is more precise (has more digits) than `vm_ver`, in which case we can't say for sure that `vm_ver` is supported.

Take Ubuntu, for example. I haven't seen an example case of this, but if `platform.linux_distribution()` returned only '16', then we can't know if it is 16.04, which we support, or 16.10, which we do not.

In your example, `vm_ver` being '6.7' and `supported_ver` being '6', yes, this should result in `vm_ver_match` being True. The IndexError will not get thrown in this case because `vm_ver` has more digits.